### ISSUE TITLE
feat: `file-extension-in-import` to consider exports map

### ISSUE THREAD

**Issue by privatenumber** [mysticatea/eslint-plugin-node#258]
When importing an extension-less path, I'd like for `file-extension-in-import` to check if the path exists in the target package's exports map. If it exists, it should not error.

eg.
```
import * as assert from 'uvu/assert';
```

Given the following [export map](https://github.com/lukeed/uvu/blob/5659d8e/package.json#L13-L34).

---

Related https://github.com/mysticatea/eslint-plugin-node/issues/255

**Comment by privatenumber** [comment]
Happy to open a PR if approved

**Comment by jaydenseric** [comment]
This issue should be labeled a bug, not an enhancement. The `node/file-extension-in-import` rule is straight-up broken, reporting false errors that are impossible to fix without using `// eslint-disable-next-line node/file-extension-in-import`. The package `exports` field is not some optional, exotic feature. It's core functionality for all the current stable Node.js versions ([claimed to be supported by `eslint-plugin-node`](https://github.com/mysticatea/eslint-plugin-node/blob/f45c6149be7235c0f7422d1179c25726afeecd83/package.json#L6)) and in use by all sorts of popular packages.

Here is a situation I run into a lot:

```js
// eslint-disable-next-line node/file-extension-in-import
import { red } from "kleur/colors";
```
